index.js

```diff
@@ -70,6 +71,9 @@ module.exports = function(schema, options) {
 
                             // Wrap with case-insensitivity
                             if (path.options && path.options.uniqueCaseInsensitive) {
+                                // Excaping some symbols to make sure that some
+                                // symbols will not be interpretated as regExp expression
+                                pathValue = escapeRegExp(pathValue);
                                 pathValue = new RegExp('^' + pathValue + '$', 'i');
```

Comment:
I think `escapeRegExp` is self-explanatory and is small enough that we can put it inline:

`pathValue = new RegExp('^' + escapeRegExp(pathValue) + '$', 'i');`